src/main/java/hudson/plugins/validating_string_parameter/ValidatingStringParameterValue.java

```diff
@@ -73,12 +76,28 @@ public BuildWrapper createBuildWrapper(AbstractBuild<?, ?> build) {
                 public Environment setUp(AbstractBuild build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
                     throw new AbortException("Invalid value for parameter [" + getName() + "] specified: " + value);
                 }
+
+                @Override
+                public Launcher decorateLauncher(AbstractBuild build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException, Run.RunnerAbortedException {
+                    throw new AbortException("Invalid value for parameter [" + getName() + "] specified: " + value);
+                }
+
             };
         } else {
             return null;
         }
     }
 
+    @Override
+    public void buildEnvironment(Run<?, ?> build, EnvVars env) {
+        if (!Pattern.matches(regex, value)) {
+            throw new Run.RunnerAbortedException();
```

Comment:
When this triggers it prints a stack trace like the following
```
hudson.model.Run$RunnerAbortedException
	at hudson.plugins.validating_string_parameter.ValidatingStringParameterValue.buildEnvironment(ValidatingStringParameterValue.java:94)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.getEnvironment(WorkflowRun.java:438)
	at org.jenkinsci.plugins.workflow.support.DefaultStepContext.get(DefaultStepContext.java:65)
	at org.jenkinsci.plugins.workflow.steps.StepDescriptor.checkContextAvailability(StepDescriptor.java:258)
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:206)
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:153)
	at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:108)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1218)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1027)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:157)
	at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:133)
	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:159)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:129)
	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
	at WorkflowScript.run(WorkflowScript:17)
        ...
```
which is not very informative IMHO.
Do you think we could print some more information on Jenkins console about why the build failed?

Reply:
@naveednu may be just add a message `"Invalid value for parameter [" + getName() + "] specified: " + value`. Also if there is a way to get the failure error message - https://github.com/naveednu/validating-string-parameter-plugin/blob/b44644e668f9b53264e1f5e636eb40051d5808e2/src/main/java/hudson/plugins/validating_string_parameter/ValidatingStringParameterDefinition.java#L57 that will be even better

Reply:
Right. But how can the message be printed to Jenkins console?
Run.RunnerAbortException() doesn't accept parameters..

Reply:
@jlelli I think we can since Run.RunnerAbortException() extends RunTimeException we can have error messages.

Reply:
@sampritavh @naveednu something like this?
https://github.com/jlelli/validating-string-parameter-plugin/commit/c76ec48b645fb5315c064bf23553e708f671c530#diff-7f2cc2852df243adfa24654a14dc3d5cR94